### PR TITLE
[java] fix a bug with nested classes if the outter class is generic

### PR DESCRIPTION
**Pull request by WinterGrascph**

### CODE REVIEW

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java [hunk 1]
```diff
@@ -436,7 +437,7 @@ public Object visit(ASTName node, Object data) {
             // Carry over the type from the declaration
             Class<?> nodeType = ((TypeNode) node.getNameDeclaration().getNode()).getType();
             // FIXME : generic classes and class with generic super types could have the wrong type assigned here
```
Comment (jsotuyod) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java]: is this fixme still valid after this added condition?

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/visitors/PMDASMVisitor.java [hunk 2]
```diff
@@ -24,7 +24,12 @@
 
     private String outerName;
 
-    private Map<String, String> packages = new HashMap<>();
+    private Map<String, String> packages = new HashMap() {
```
Comment (jsotuyod) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/visitors/PMDASMVisitor.java]: what's the point of this override?

File: pmd-java/src/test/java/net/sourceforge/pmd/typeresolution/testdata/FieldAccessGenericSimple.java [hunk 3]
```diff
@@ -53,8 +53,10 @@ void foo(GenericClass<Integer, Character> param) {
     }
 
     public class Nested extends GenericSuperClassA<Long> {
+        //T outterParameterField;
         void foo() {
             fieldA = new Long(0);
+            //outterParameterField = null;
```
Comment (jsotuyod) [inline pmd-java/src/test/java/net/sourceforge/pmd/typeresolution/testdata/FieldAccessGenericSimple.java]: why are these commented out?

**Comment by jsotuyod** [comment]
Is this a fix for #496?

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java (hunk 1 above)
Comment (WinterGrascph) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java]: The added condition is there by mistake.

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/visitors/PMDASMVisitor.java (hunk 2 above)
Comment (WinterGrascph) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/visitors/PMDASMVisitor.java]: This override is there by mistake.

File: pmd-java/src/test/java/net/sourceforge/pmd/typeresolution/testdata/FieldAccessGenericSimple.java (hunk 3 above)
Comment (WinterGrascph) [inline pmd-java/src/test/java/net/sourceforge/pmd/typeresolution/testdata/FieldAccessGenericSimple.java]: Those are there by mistake.

**Comment by WinterGrascph** [comment]
Yes, it seems like it's a fix for #496 .

**Pull request merged**